### PR TITLE
Initial changes to try to fix cata beta.

### PR DESCRIPTION
**Pull request by Flamanis**
Initial wave of changes to deal with Cata beta and try to fix issues.

Just changes to the checks for the most part. A couple of things for setup, like LibOpenRaid or the spell_cache.

Fix for `GetTalentTabInfo` giving more information in cata beta than wrath/era have.

### CODE REVIEW

File: Libs/DF/fw.lua [hunk 1]
```diff
@@ -188,15 +188,6 @@ function DF.IsNonRetailWowWithRetailAPI()
...
 DF.IsWotLKWowWithRetailAPI = DF.IsNonRetailWowWithRetailAPI -- this is still in use
 
----return true if the version of wow the player is playing is the shadowlands
-function DF.IsShadowlandsWow()
-    local _, _, _, buildInfo = GetBuildInfo()
-    if (buildInfo < 100000 and buildInfo >= 90000) then
-        return true
-    end
-	return false
-end
```
Comment (Flamanis) [inline Libs/DF/fw.lua:198]: Function is already defined like 20 lines above. This is extra

File: Libs/DF/spells.lua [hunk 2]
```diff
@@ -1415,8 +1415,8 @@ elseif (DF.IsWotLKWow()) then
 	}
 	DF.FeastIDs = {}
 	DF.RuneIDs = {}
-
-elseif (DF.IsClassicWow()) then
+--~Cata temp
+elseif (DF.IsClassicWow() or DF.IsCataWow()) then
```
Comment (Flamanis) [inline Libs/DF/spells.lua:1419]: Marked temporary setup items to deal with objects not existing in cata beta as `~Cata temp` to make for finding them easier.

File: Libs/LibOpenRaid/lib.xml [hunk 3]
```diff
@@ -7,4 +7,5 @@
 	<Script file="ThingsToMantain_Shadowlands.lua" />
 	<Script file="ThingsToMantain_Wrath.lua" />
 	<Script file="ThingsToMantain_Era.lua" />
+	<Script file="ThingsToMantain_Cata.lua" />
```
Comment (Flamanis) [inline Libs/LibOpenRaid/lib.xml:10]: File is just a direct copy of the wrath one for now

File: frames/anime.lua [hunk 4]
```diff
@@ -234,7 +234,7 @@ function _detalhes.PlayBestDamageOnGuild (damage)
 	Portrait.scale:SetStartDelay (0)
 	Portrait.scale:SetEndDelay (0)
 
-	if (DetailsFramework.IsDragonflight() or DetailsFramework.IsWotLKWowWithRetailAPI()) then
+	if (DetailsFramework.IsDragonflight() or DetailsFramework.IsNonRetailWowWithRetailAPI()) then
```
Comment (Flamanis) [inline frames/anime.lua:237]: Change the function name to the 'correct' one. Just makes for finding these functions easier.

File: functions/spells.lua [hunk 5]
```diff
@@ -533,8 +533,8 @@ do
 
 
 		}
-
-	elseif (DetailsFramework.IsWotLKWow()) then
+	--~Cata temp
+	elseif (DetailsFramework.IsWotLKWow() or DetailsFramework.IsCataWow()) then
 		_detalhes.SpecSpellList = { --~spec
```
Comment (Flamanis) [inline functions/spells.lua:539]: Another temp, need to have access to beta to figure out the proper list.

File: core/gears.lua [hunk 6]
```diff
@@ -2526,7 +2526,12 @@ if (DetailsFramework.IsWotLKWow()) then
 		for i = 1, (MAX_TALENT_TABS or 3) do
 			if (i <= numTabs) then
 				--tab information
-				local name, iconTexture, pointsSpent, fileName = GetTalentTabInfo (i)
+				local id, name, description, iconTexture, pointsSpent, fileName
+				if DetailsFramework.IsCataWow() then
+					id, name, description, iconTexture, pointsSpent, fileName = GetTalentTabInfo(i)
```
Comment (Flamanis) [inline core/gears.lua:2531]: According to BlizzardInterfaceCode the return values for `GetTalentTabInfo` on cata beta are `local id, name, description, icon, pointsSpent, background, previewPointsSpent, isUnlocked` This change just pulls the same values we were originally due to the new argument count.

**Comment by Toodo** [comment]
hello thanks for trying to update it i gave you some lua error i got when enter wow. but overall it working now :)

**Comment by Flamanis** [comment]
> hello thanks for trying to update it i gave you some lua error i got when started wow. but overall it working now :) i use this build [c1fa712](https://github.com/Tercioo/Details-Damage-Meter/commit/c1fa712b9130a04effee7e8e3d45b37085aba3f4)

You're missing localization files

**Comment by Toodo** [comment]
okey so what do i need to do? im not good at this just wanted the addon to work so i gave your fix a try ^^

**Pull request merged**